File: notifications-polyglot/java/src/main/java/com/example/notification/adapter/EmailAdapter.java
package com.example.notification.adapter;

import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

@Slf4j
@Component
public class EmailAdapter implements ChannelAdapter {

    @Override
    public boolean supports(String channel) {
        return "email".equalsIgnoreCase(channel);
    }

    @Override
    public void send(String recipientId, String message) {
        log.info("[Email] Sending to {}: {}", recipientId, message);
        try {
            Thread.sleep(200); // Simulate latency
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
        }
    }
}

File: notifications-polyglot/kotlin/src/main/kotlin/com/example/notification/adapter/EmailAdapter.kt
package com.example.notification.adapter

import kotlinx.coroutines.delay
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

@Component
class EmailAdapter : ChannelAdapter {
    private val logger = LoggerFactory.getLogger(EmailAdapter::class.java)

    override fun supports(channel: String): Boolean {
        return "email".equals(channel, ignoreCase = true)
    }

    override fun send(recipientId: String, message: String) {
        logger.info("[Email] Sending to {}: {}", recipientId, message)
        Thread.sleep(200) // Simulate latency
    }
}
